xbmc/settings/lib/SettingsManager.cpp

```diff
@@ -1248,7 +1248,9 @@ void CSettingsManager::ResolveReferenceSettings(std::shared_ptr<CSettingSection>
         else
         {
           referencedSetting = itReferencedSetting->second.setting;
-          m_settings.erase(FindSetting(referenceSetting->GetId()));
+          auto itReferencedSetting2 = FindSetting(referenceSetting->GetId());
```

Comment:
This is ok for solving the crash if the same setting is used more than twice (in which case it has already been removed from `m_settings`. There are still issues however with reference settings and enable/visibility conditions which I'm trying to solve.